homework/btvn-2/ValidParentheses.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+class SimpleStack
+{
+    protected $queue = [];
+
+    public function push($data)
+    {
+        var_dump($this->queue);
+        $this->queue[] = $data;
+    }
+
+    public function pop()
+    {
+        var_dump($this->queue);
+        array_pop($this->queue);
+    }
+
+    public function isEmpty()
+    {
+        return empty($this->queue);
+    }
+
+    public function peek()
+    {
+        if (!empty($this->queue)) {
+            return $this->queue[count($this->queue) - 1];
+        } else {
+            return false;
+        }
+    }
+}
+
+
+class ValidParentheses {
+
+    /**
+     * @param String $s
+     * @return Boolean
+     */
+    function isValid($s) {
+        $queue = new SimpleStack();
+        $arrayS = str_split($s);
+
+        foreach ($arrayS as $singleString)
+        {
+            if ($singleString == '(' or $singleString == '{' or $singleString == '[') {
+                $queue->push($singleString);
+            }
+
+            if ($singleString == '}' and $queue->peek() != '{') {
```

Comment:
this part is not DRY, can we combine all 3 conditions? consider make a separated function, https://github.com/quanghoang33/gen-6/pull/18/files#diff-3d2ef42514a8452162796ca6bd825b224e4a2443793769bd33be135184b74bf4R10